processor/routingprocessor/internal/common/logger.go

```diff
@@ -0,0 +1,64 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/routingprocessor/internal/common"
+
+import (
+	"fmt"
+
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/telemetryquerylanguage/tql"
+)
+
+type TQLLogger struct {
+	logger *zap.Logger
+}
+
+func NewTQLLogger(logger *zap.Logger) TQLLogger {
+	return TQLLogger{
+		logger: logger,
+	}
+}
+
+// WithFields creates a new logger that will include the specified fields
+// in all subsequent logs in addition to fields attached to the context
+// of the parent logger. Note that fields are not deduplicated.
+func (tqll TQLLogger) WithFields(fields map[string]any) tql.Logger {
```

Comment:
I think the Go conventions state the receiver's name should have one letter.

edit: found a reliable source for this: https://github.com/golang/go/wiki/CodeReviewComments#receiver-names

```suggestion
func (t TQLLogger) WithFields(fields map[string]any) tql.Logger {
```